### PR TITLE
update circuitbreaker document 

### PR DESCRIPTION
**Pull request by atawLee**
#1776 

Update documentation for the CircuitBreaker's BreakDurationGenerator

## Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [x]  I have included unit tests for the issue/feature
- [x]  I have successfully run a local build

### CODE REVIEW

File: docs/strategies/circuit-breaker.md [hunk 1]
```diff
@@ -88,6 +98,7 @@ new ResiliencePipelineBuilder<HttpResponseMessage>().AddCircuitBreaker(optionsSt
 | `MinimumThroughput` | 100                                                                        | The minimum number of actions that must occur in the circuit within a specific time slice. |
 | `SamplingDuration`  | 30 seconds                                                                 | The time period over which failure ratios are calculated.                                  |
 | `BreakDuration`     | 5 seconds                                                                  | The time period for which the circuit will remain broken/open before attempting to reset.  |
+| `BreakDurationGenerator`| `null`                                                            | Enables adaptive adjustment of break duration based on the current state of the circuit.  |
```
Comment (martincostello) [inline docs/strategies/circuit-breaker.md:101]: Would you mind re-aligning the table please?

**Review by martincostello — COMMENTED** [review]
Going through https://github.com/App-vNext/Polly/pull/1776#issuecomment-1805758347 looks like just a diagram is needed to complete the suggestions.

**Comment by peter-csala** [comment]
@atawLee Do you need some help with the diagram? I can create one if you wish.

**Comment by martincostello** [comment]
`BreakDurationGenerator` needs adding to the [spellcheck dictionary](https://github.com/atawLee/Polly/blob/breakDurationDoc/.github/wordlist.txt).

File: docs/strategies/circuit-breaker.md [hunk 2]
```diff
@@ -308,6 +308,48 @@ sequenceDiagram
     P->>C: Propagates failure
 ```
 
+#### Complex: dynamic break duration sequence diagram
+
+This sequence diagram illustrates the behavior of a circuit breaker using a BreakDurationGenerator. The generator dynamically calculates the break duration based on specific criteria, such as the number of failures
```
Comment (martincostello) [inline docs/strategies/circuit-breaker.md:313]:
```suggestion
This sequence diagram illustrates the behavior of a circuit breaker using a `BreakDurationGenerator`. The generator dynamically calculates the break duration based on specific criteria, such as the number of failures:
```

**Review by peter-csala — APPROVED** [review]

**Review by martincostello — APPROVED** [review]

**Pull request merged**